src/validator.ts

```diff
@@ -680,6 +682,51 @@ function doIntrinsicGetAZs(ref: any, key: string){
 
 }
 
+function doIntrinsicSelect(ref: any, key: string){
+  let toGet = ref[key];
+  if(!Array.isArray(toGet) || toGet.length < 2) {
+    addError('crit', "Fn::Select only supports an array of two elements", placeInTemplate, "Fn::Select");
+    return 'INVALID_SELECT';
+  }
+  if (toGet[0] === undefined || toGet[0] === null) {
+     addError('crit', "Fn::Select first element cannot be null or undefined", placeInTemplate, "Fn::Select");
+     return 'INVALID_SELECT';
+  }
+
+  let index = parseInt(toGet[0]);
+  if (typeof index !== 'number' || isNaN(index)) {
+    addError('crit', "First element of Fn::Select must be a number", placeInTemplate, "Fn::Select");
+    return 'INVALID_SELECT';
+  }
+  if (toGet[1] === undefined || toGet[1] === null) {
+     addError('crit', "Fn::Select Second element cannot be null or undefined", placeInTemplate, "Fn::Select");
+     return 'INVALID_SELECT';
+  }
+
+  let list = toGet[1]
+  if (!Array.isArray(list)) {
```

Comment:
It looks like CommaDelimitedList should be supported.
```
    Type: CommaDelimitedList
    Default: "10.0.48.0/24, 10.0.112.0/24, 10.0.176.0/24"
```
https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-select.html

Reply:
I'll be sure to get that in there too.

Reply:
This should be automatically converted using this lookup https://github.com/martysweet/cfn-lint/blob/master/data/aws_parameter_types.json. So should just be a case of using the array.

There may be some validation issues here https://github.com/martysweet/cfn-lint/blob/master/src/validator.ts#L247-L263 which don't catch invalid CommaDelimitedList provided lists. If this turns into a lot of effort we can split improving that into another PR.

Reply:
Hmm, how would I go about getting that array? I'm not seeing it in `workingInput`, but maybe I'm looking in the wrong place. ` workingInput['Parameters'][parameterName]['Attributes']['Ref']` is just returning the original string. I could just do a `.split(',')` and `.trim()` each value in the array from that. But I feel like there is a better way to do that. :)

Reply:
Here's what I added to `assignParametersOutput` just after it checks allowedValues
```ts
        if(parameter['Type'] === "CommaDelimitedList" && typeof parameterValue === 'string') {
            parameterValue = parameterValue.split(',').map(x => x.trim());
            parameterValue.forEach(val => {
              if (val === ""){
                addError('crit', `Parameter ${parameterName} contains a CommaDelimitedList where the number of commas appears to be equal or greater than the list of items.`, ['Parameters', parameterName], "Parameters");
              }
            })
        }
```